src/main/java/ui/Ui.java

```diff
@@ -1,52 +1,104 @@
 package ui;
 
+import language.Language;
+import language.LanguageFactory;
+import language.Languages;
 import move.Moves;
 
 import java.io.InputStream;
 import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Scanner;
 
 public class Ui {
 
+    final static private String NEWLINE = "";
+    private Language language;
     private PrintStream output;
     private Scanner input;
 
-    public Ui(PrintStream output, InputStream input) {
+    public Ui(PrintStream output, InputStream input, Language language) {
         this.output = output;
         this.input = new Scanner(input);
+        this.language = language;
```

Comment:
No need to pass in language, you could just set it to `new English()` as default in the constructor.

Reply:
Right!